core/templates/App.ts

```diff
@@ -79,6 +79,32 @@ angular.module('oppia').config([
   function(
       $compileProvider, $cookiesProvider, $httpProvider,
       $interpolateProvider, $locationProvider, $provide) {
+    var ugs = new UpgradedServices();
+    // We need to provide these services separately since they are
+    // used in the directives imported in this file and cannot be
+    // injected before bootstrapping of oppia module.
+    var servicesToProvide = [
```

Comment:
How will this list be kept up-to-date?

Reply:
Yes this list has to be kept up to date.

We need to provide these services in both prod and test since bootstrapping is not done when these services are required in directives. It will be kept up to date by anyone who upgrades a new service and sees that the tests fail or localhost server is not working due to services being unavailable.